PlaybackLab/ControlsLockScreenSampleApp/ControlsLockScreenSampleApp/Views/VideoViewController.swift

```diff
@@ -0,0 +1,144 @@
+      
+      player = OOOoyalaPlayer(pcode: option.getPcode(), domain: option.getDomain(), embedTokenGenerator: option.getEmbedTokenGenerator()!, options: options)
+    }
+    else {
+      player = OOOoyalaPlayer(pcode: option.getPcode(), domain: option.getDomain())
+    }
+    
+    player.setEmbedCode(option.getEmbedCode())
+    
+    let remoteCommandCenter: MPRemoteCommandCenter = MPRemoteCommandCenter.shared()
+    remoteCommandCenter.playCommand.addTarget { (event) -> MPRemoteCommandHandlerStatus in
+      self.player.play()
+      return MPRemoteCommandHandlerStatus.success
+    }
+    
+    remoteCommandCenter.pauseCommand.addTarget(handler: { (event) -> MPRemoteCommandHandlerStatus in
+      self.player.pause()
+      return MPRemoteCommandHandlerStatus.success
+    })
+    
+    remoteCommandCenter.skipForwardCommand.preferredIntervals = [15]
+    remoteCommandCenter.skipForwardCommand.addTarget(handler: { (event) -> MPRemoteCommandHandlerStatus in
+      self.player.seek(self.player.playheadTime() + 15.0)
+      return MPRemoteCommandHandlerStatus.success
+    })
+    
+    remoteCommandCenter.skipBackwardCommand.preferredIntervals = [15]
+    remoteCommandCenter.skipBackwardCommand.addTarget(handler: { (event) -> MPRemoteCommandHandlerStatus in
+      self.player.seek(self.player.playheadTime() - 15.0)
+      return MPRemoteCommandHandlerStatus.success
+    })
+    
+    // Set properties of the asset to be shown in the screen (no required).
+    var nowPlayingInfo: [String : Any] = [MPMediaItemPropertyTitle: option.getTitle(),
+                                          MPMediaItemPropertyArtist: "Ooyala",
+                                          MPMediaItemPropertyAlbumTitle: "Controls Lock Screen"
+    ]
+    
+    // Set albumArt to show in the screen if image is available
+    // The albumArt needs to be an UIImage, the image can be added in the assets of the project or retrieve the image using an URL
+    if let imageData: NSData = NSData(contentsOf: option.getThumbnailURL()), let image: UIImage = UIImage(data: imageData as Data) {
+      let albumArt: MPMediaItemArtwork = MPMediaItemArtwork(boundsSize: image.size, requestHandler: { (size) -> UIImage in return image })
+      nowPlayingInfo[MPMediaItemPropertyArtwork] = albumArt
+    }
+    
+    MPNowPlayingInfoCenter.default().nowPlayingInfo = nowPlayingInfo
+    
+    NotificationCenter.default.addObserver(self, selector: #selector(applicationDidEnterBackground(_:)), name: NSNotification.Name.UIApplicationDidEnterBackground, object: nil)
+    NotificationCenter.default.addObserver(self, selector: #selector(playerStateChange(_:)), name: NSNotification.Name.OOOoyalaPlayerStateChanged, object: nil)
+  }
+  
+  override func viewWillAppear(_ animated: Bool) {
+    super.viewWillAppear(animated)
+    UIApplication.shared.beginReceivingRemoteControlEvents()
+    becomeFirstResponder()
+    
+    player.play()
+  }
+  
+  override func viewWillDisappear(_ animated: Bool) {
```

Comment:
I think we need to sync offline to talk about the lifetime of the player. I see you destroy this player when the VC is no visible anymore which is good. But when playing it from the Background what player instance does it use? Not sure about how this works.

Reply:
The player is not destroyed when the app goes to background. Only when you go back to the options (assets).